xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp

```diff
@@ -1243,6 +1249,11 @@ DemuxPacket* CDVDDemuxFFmpeg::Read()
   return pPacket;
 }
 
+DemuxPacket* CDVDDemuxFFmpeg::Read()
+{
+  return ReadInternal(false);
```

Comment:
What about keeping a boolean state member variable to specify if the packet should be kept or not instead of adding the bool to the argument? Should only be a matter of setting it to true before calling read I guess

Reply:
I would not do that: Reason, you will always need two function calls. From API usage this complicates things. I would keep it as is.

If stuff should be really obvious. Adding and additional Peek() method would be my choice, so it is clear that Read does Pop while Peek does peek or something.

Also I think, can easily be a follow up.